src/test/java/org/sonar/plugins/checkstyle/internal/ChecksTest.java

```diff
@@ -452,7 +452,7 @@ else if (AbstractFileSetCheck.class.isAssignableFrom(clss)) {
         final PropertyDescriptor[] map = PropertyUtils.getPropertyDescriptors(clss);
 
         for (PropertyDescriptor p : map) {
-            if (p.getWriteMethod() != null) {
+            if (p.getWriteMethod() != null && !p.getWriteMethod().isAnnotationPresent(Deprecated.class)) {
```

Comment:
@romani We don't hide deprecated properties in main repo. We should do that in both places?

Reply:
Can you explain me what is reason why we have should show deprecated users properties ?
Example: https://checkstyle.org/config_javadoc.html#JavadocMethod_Properties (ignoreMethodNamesRegex). If no good reason , we should remove them, to make breaking compatibility actual.

We should remove properties as clear signal that they are not effective any more and to not give user any fake filling that Check is working as before.
user should not get to know in hard way that property is not working any more, we should fail fast.

In main repo, we did not have Deprecated properties, if we deprecate we did removal of methods and removal from documentation.

Reply:
@romani 
>  If no good reason , we should remove them, to make breaking compatibility actual.

I thought this was the reason to mark them deprecated. To not break compatibility completely in 1 go, otherwise I can just remove the methods completely which would break every config using the module and require an override for our CI.

> In main repo, we did not have Deprecated properties, if we deprecate we did removal of methods and removal from documentation.

So remove all deprecated properties from main repo? If you agree, I will create an issue.

Reply:
If we break compatibility functionally it is the same as breaking it in config. If both is together, users will have to spend time with Checkstyle only ones. I almost 99% sure, nobody activate new Checks right after release. Majority of users do not care about our releases till something inconvenient happens on their side and and they search for fix by upgrade to latest version.

So, I do not think that migration process is required for properties of Check. It might be ok make it in few steps if we need it.

Please create new issue, it will be one more breaking change.